src/GraphQL.Dynamic/LiteralGraphType/LiteralGraphTypeHelpers.cs

```diff
@@ -0,0 +1,158 @@
+    internal delegate object GetValueFn(ResolveFieldContext context);
+
+    internal static class LiteralGraphTypeHelpers
+    {
+        private static readonly Regex SanitizeTypeNameRegex = new Regex(@"[:\/.]");
+
+        private static ConcurrentDictionary<string, FieldTypeResolver> MemberFieldTypeResolverMappings { get; set; } = new ConcurrentDictionary<string, FieldTypeResolver>();
+        private static ConcurrentDictionary<string, FieldType> MemberFieldTypeMappings { get; set; } = new ConcurrentDictionary<string, FieldType>();
+
+        private static ConcurrentDictionary<LiteralGraphTypeMemberInfoType, Type> LiteralGraphTypeMemberInfoTypeMappings { get; set; } = new ConcurrentDictionary<LiteralGraphTypeMemberInfoType, Type>();
+        private static ConcurrentDictionary<LiteralGraphTypeMemberInfoType, FieldTypeResolver> LiteralGraphTypeMemberInfoFieldTypeResolverMappings { get; set; } = new ConcurrentDictionary<LiteralGraphTypeMemberInfoType, FieldTypeResolver>();
+
+        private readonly static GraphTypesLookup _graphTypesLookup = new GraphTypesLookup();
+
+        static LiteralGraphTypeHelpers()
+        {
+            PopulateMappings();
+        }
+
+        private static void PopulateTypeMapping(LiteralGraphTypeMemberInfoType type, Type actualType, FieldTypeResolver resolver)
+        {
+            LiteralGraphTypeMemberInfoTypeMappings.TryAdd(type, actualType);
+            LiteralGraphTypeMemberInfoFieldTypeResolverMappings.TryAdd(type, resolver);
+        }
+
+        private static void PopulateMappings()
+        {
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.String, typeof(string), CreateFieldType<StringGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.Int, typeof(int), CreateFieldType<IntGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.Long, typeof(long), CreateFieldType<IntGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.Float, typeof(float), CreateFieldType<FloatGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.Double, typeof(double), CreateFieldType<FloatGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.Boolean, typeof(bool), CreateFieldType<BooleanGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.Guid, typeof(Guid), CreateFieldType<IdGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.DateTime, typeof(DateTimeOffset), CreateFieldType<DateGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.DateTimeOffset, typeof(DateTimeOffset), CreateFieldType<DateGraphType>);
+        }
+
+        internal static Type GetPrimitiveGraphType(string name) => _graphTypesLookup[name]?.GetType();
+
+        internal static string GenerateRemoteTypeName(string remoteLocation, string name) => SanitizeTypeNameRegex.Replace($"{remoteLocation}.{name}", "_");
```

Comment:
putting the full url in front of every remote type gives a very verbose schema. the docs look very busy as well. can't we find a simpler way to solve this? e.g. using the moniker or an intelligent solution which only triggers for conflicting types.